### DISCUSSION TITLE
Conversion from DOCX to PDF loses format and text

### DISCUSSION THREAD

**Discussion by manoharpview** [gotenberg/gotenberg#967] [Q&A]
Hello, 

I tried converting attached docx file to pdf for rendering in UI for preview, however, conversion didn't go well. When i converted same using liberoffice conversion was good and as i expected. `Did i missing anything as prerequisite or some setting?  
`
I am using : `gotenberg/gotenberg:8`
No additional  or custom config changes, just using default image

[converted-using-liberoffice.pdf](https://github.com/user-attachments/files/16933020/converted-using-liberoffice.pdf)
[input.docx](https://github.com/user-attachments/files/16933025/input.docx)
[converted-using-gotenberg.pdf](https://github.com/user-attachments/files/16933035/converted-using-gotenberg.pdf)

**Comment by gulien** [comment]
Hello @manoharpview 👋 

Fonts look different, I guess some of your local fonts are not installed in the image - see https://gotenberg.dev/docs/configuration#fonts for installing custom one!

**Reply by manoharpview** [reply]
@gulien , I dont have any fonts in my local, or at least i dont know if there any. How did i check for these fonts and copy them .

**Reply by gulien** [reply]
> I dont have any fonts in my loca

Of course you have, otherwise LibreOffice won't even work lol

> How did i check for these fonts and copy them 

Using an [online tool](https://www.i2pdf.com/extract-fonts-from-pdf) with your file [converted-using-liberoffice.pdf](https://github.com/user-attachments/files/16933020/converted-using-liberoffice.pdf), I found these fonts:  `Gill Sans.ttf` and 
`Gill Sans_2.ttf`.

Using the same process, exract these fonts from the PDF and install them in a custom Docker image as explained here: https://gotenberg.dev/docs/configuration#fonts

**Reply by manoharpview** [reply]
@gulien , i copied ttf files to `/usr/local/share/fonts` as mentioned in documentation, but, i still cant get pdf converted properly. Also,  given that gotenberg uses liberoffice in background aren't these fonts already available. 

Thanks for quick reply.

**Reply by gulien** [reply]
Could you show me the resulting PDF?

There is only a selected range of fonts in the default Docker image, that’s why adding the selected should have resolved the issue. Your local fonts are not available in the image by default.

**Reply by manoharpview** [reply]
[converted-with-new-fonts.pdf](https://github.com/user-attachments/files/16958676/converted-with-new-fonts.pdf)

here is the pdf after conversion

also, i think it will be difficult to add fonts without knowing, possible styles that user uploads might contain. Is there away to add all fonts that liberoffice supports ?

**Reply by gulien** [reply]
He doesn't look like the PDF you've converted from LibreOffice UI 🤔 Which version of LibreOffice are you using? I wonder if all required fonts are available. You may try to mount all your fonts available in your system.

LibreOffice does not have a limited range of supported fonts, and I cannot include all fonts from the universe in the Docker image 😄
